src/Workspaces/CSharp/Portable/Recommendations/CSharpRecommendationService.cs

```diff
@@ -75,8 +75,13 @@ internal class CSharpRecommendationService : AbstractRecommendationService
                 // Script and interactive
                 return GetSymbolsForGlobalStatementContext(context, cancellationToken);
             }
-            else if (context.IsAnyExpressionContext || context.IsStatementContext)
+            else if (context.IsAnyExpressionContext || 
+                     context.IsStatementContext || 
+                     context.SyntaxTree.IsDefiniteCastTypeContext(context.Position, context.LeftToken, cancellationToken))
             {
+                // GitHub #717: With automatic brace completion active, typing '(' produces "()", which gets parsed as
+                // as cast. The user might be trying to type a parenthesized expression, so even though a cast
+                // is a type-only context, we'll show all symbols anyway.
```

Comment:
In the case from the bug:
```C#
        for (int i = 0; i< 5; i++)
        {
            var x = ()
            var y = 1;
        }
```
The "()" is parsed as a parenthesized expression, but when the first character is inserted inside the parentheses it then switches to a CastExpression.

I think "i" would have been in the list inside the parentheses if completion was invoked explicitly, but not once a character (the "i") is actually typed.

This is a problem with the comment but also with the unit test that was added. Can you also add another unit test for the case when the i is already in the parentheses?